omx/gstomx.c

```diff
@@ -1370,7 +1389,23 @@ gst_omx_port_acquire_buffer (GstOMXPort * port, GstOMXBuffer ** buf)
 
   GST_DEBUG_OBJECT (comp->parent, "%s port %u has pending buffers",
       comp->name, port->index);
-  _buf = g_queue_pop_head (&port->pending_buffers);
+  if (*buf) {
```

Comment:
Consider adding comment to explain this piece of logic

Reply:
The comment at the beginning of the function explains a little bit, so that can be moved and also add doxygen documentation.